controller/common/lut.py

```diff
@@ -0,0 +1,78 @@
+from typing import List, Union
+
+import numpy as np
+from scipy import interpolate
+
+
+class LUT:
+    """
+    Class for performing look-up table interpolation.
+
+    Methods:
+        __init__: Initializes the LUT object with lookup table data and interpolation method.
+        __call__: Calls the interpolation method with the given input.
+    """
+
+    def __init__(
+        self,
+        lookup_table: Union[List[List[int | float]], np.ndarray],
+        interpolation_method: str = "linear",
+    ):
```

Comment:
In Python 3.10, using `Union` is equivalent to using `|`. For example, `Union[int, float]` is the same as `int | float`.

I also created a types module in the controller repository. I created a type `Scalar` for `int | float`:

```python
from controller.common.types import Scalar
```

Also when typing for numpy arrays, one should use the `NDArray` type instead of `np.ndarray`:

```python
from numpy.typing import NDArray
```

Reply:
updated

```
    def __init__(
        self,
        lookup_table: List[List[Scalar]]|NDArray,
        interpolation_method: str = "linear",
    ):
```